### PR TITLE
Cover Hover Previews

### PR DESCRIPTION
**Pull request by MasterBugPatch**
When a user hovers over a preview of a cover, their cover will turn into
it until their mouse leaves the preview.

### CODE REVIEW

**Comment by peppy** [comment]
This project is using UK/Australian spelling by default, so please remove the spelling commit (it shouldn't be grouped together with other changes anyway).

File: resources/assets/js/jsx/profile_page.jsx [hunk 1]
```diff
@@ -53,16 +53,23 @@
 					dataType: 'json'
 			})
 			.done(function(userData) {
+				this.props.user.cover.id = userData.data.cover.id;
```
Comment (nanaya) [inline resources/assets/js/jsx/profile_page.jsx]: this and the following line shouldn't be necessary. Additionally, props may never be modified.

File: resources/assets/js/jsx/profile_page.jsx [hunk 2]
```diff
@@ -53,16 +53,23 @@
...
 
 		changeMode: function(e) {
 			e.preventDefault();
-
 			this.setState({ mode: e.target.getAttribute('data-mode') });
 		},
-
+		
+		previewCover: function(_e, p) {//handles hover event
+			var obj = this.props.user;
```
Comment (nanaya) [inline resources/assets/js/jsx/profile_page.jsx]: props.user only used for initial state. Use state.user instead.

File: resources/assets/js/jsx/profile_page/header.jsx [hunk 3]
```diff
@@ -25,6 +25,15 @@
 			$(document).trigger('profile:cover:select', this.props.name);
 		},
 
+		onMouseEnter: function() {
+			$(document).trigger('profile:previewcover', this.props.name);
+		},
+		
+		onMouseLeave: function() {
+			if(this.props.selectedName != this.props.name){
+			$(document).trigger('profile:previewcover', this.props.selectedName);
```
Comment (nanaya) [inline resources/assets/js/jsx/profile_page/header.jsx]: this doesn't return custom profile image back.

File: resources/assets/js/jsx/profile_page/header.jsx [hunk 4]
```diff
@@ -25,6 +25,15 @@
 			$(document).trigger('profile:cover:select', this.props.name);
 		},
 
+		onMouseEnter: function() {
+			$(document).trigger('profile:previewcover', this.props.name);
+		},
+		
+		onMouseLeave: function() {
+			if(this.props.selectedName != this.props.name){
```
Comment (nanaya) [inline resources/assets/js/jsx/profile_page/header.jsx]: never use loose comparison (`!=`, `==`) in javascript.

File: resources/assets/js/jsx/profile_page.jsx [hunk 5]
```diff
@@ -94,6 +102,7 @@
 			this.unlistenAll();
 			$(document).on('profile:cover:select', this.changeCoverDefault);
 			$(document).on('profile:updated', this.updateData);
+			$(document).on('profile:previewcover', this.previewCover);
```
Comment (nanaya) [inline resources/assets/js/jsx/profile_page.jsx]: should be `profile:cover:preview` to follow other events naming pattern.

**Comment by MasterBugPatch** [comment]
Alright, I'll have all of these fixed by tonight. Thanks for the feedback ^.^

**Comment by TheHowl** [comment]
Some of the spelling fixes are actually right though.

http://dictionary.cambridge.org/dictionary/english/repository (search for respository, it won't find it)

http://dictionary.cambridge.org/dictionary/english/guarantee

File: resources/assets/js/jsx/profile_page.jsx (hunk 1 above)
Comment (MasterBugPatch) [inline resources/assets/js/jsx/profile_page.jsx]: Yes but for some reason their was an error where when a user selects a cover, then re-enters a different box the selected cover will go back to its previous one. This solved that for me.

Comment (nanaya) [inline resources/assets/js/jsx/profile_page.jsx]: that's probably because you use props.user instead of state.user below.

**Pull request closed without merging**